packages/core/src/awsService/ec2/sshKeyPair.ts

```diff
@@ -28,10 +29,25 @@ export class SshKeyPair {
     }
 
     public static async getSshKeyPair(keyPath: string, lifetime: number) {
+        SshKeyPair.assertValidKeypath(
+            keyPath,
+            `ec2: unable to generate key outside of global storage in path ${keyPath}`
+        )
         await SshKeyPair.generateSshKeyPair(keyPath)
         return new SshKeyPair(keyPath, lifetime)
     }
 
+    private static isValidKeyPath(keyPath: string): boolean {
+        const relative = path.relative(globals.context.globalStorageUri.fsPath, keyPath)
+        return relative !== undefined && !relative.startsWith('..') && !path.isAbsolute(relative)
```

Comment:
Something like `keyPath.length > 4` also is a simple but effective way to avoid accidental deletion, especially when deleting a directory whose path would never be shorter than, say, 10 chars.